xclim/indices/run_length.py

```diff
@@ -451,11 +452,15 @@ def coord_transform(out, da):
         return out
 
     # general method to get indices (or coords) of first run
-    def get_out(d):
-        dmax_ind = d.argmax(dim=dim)
+    def get_out(da):
+        if first is False:
+            da = da[{dim: slice(None, None, -1)}]
+        dmax_ind = da.argmax(dim=dim)
         # If `d` has no runs, dmax_ind will be 0: We must replace this with NaN
-        out = dmax_ind.where(dmax_ind != d.argmin(dim=dim))
-        out = coord_transform(out, d)
+        out = dmax_ind.where(dmax_ind != da.argmin(dim=dim))
+        if first is False:
+            out = da[dim].size - out - 2
+        out = coord_transform(out, da)
         return out
```

Comment:
I'm not saying this should be implemented as it is. Having a `first` variable used in `first_run` to switch between `first_run` and `last_run` modes is not ideal. I just wanted to show how this can be done, and get some ideas on how to organize things. Should I make a parent function to `first_run` and `last_run`, e.g. `first_or_last_run` ?